packages/marketplace-management/src/components/ui/users/edit-user.tsx

```diff
@@ -45,13 +43,35 @@ export const formFields: Record<FieldType, FormFieldInfo> = {
   },
 }
 
+export const sortAddRemoveGroups = (editingUser: UserModel, groupIds: string[]) => {
+  const currentGroups = editingUser.groups ?? []
+  const removeIds = currentGroups.filter((group) => !groupIds.find((groupId) => groupId === group))
+  const addIds = groupIds.filter((group) => !currentGroups.find((groupId) => groupId === group))
+
+  return {
+    removeIds,
+    addIds,
+  }
+}
+
 export const onHandleSubmit = (handleOnClose: () => void, onRefetchData: () => void, editingUser?: UserModel) => async (
   params: UpdateUserModel,
 ) => {
-  const { name, groupIds } = params
-  const updateUserRes = await updateUser({ name, groupIds }, editingUser?.id || '')
+  const { groupIds } = params
+  const userId = editingUser?.id
+  if (!groupIds || !editingUser || !userId) return null
+
+  const { removeIds, addIds } = sortAddRemoveGroups(editingUser, groupIds)
+  const totalUpdates = removeIds.length + addIds.length
+
+  const updateUserRes = await Promise.all([
+    ...removeIds.map((id) => removeMemberFromGroup({ id, userId })),
+    ...addIds.map((id) => addMemberToGroup({ id, userId })),
```

Comment:
Batch endpoint?

Reply:
There should be... but there isn't...

In practice, there are only like 5 groups so it's not something to get too worried about!

Reply:
ahhh ok fair 👍